app/assets/javascripts/admin/addon/components/admin-config-areas/components.gjs

```diff
@@ -45,19 +93,428 @@ export default class AdminConfigAreasComponents extends Component {
+        },
+      });
+    } catch (error) {
+      this.toasts.error({
+        duration: 5000,
+        data: {
+          message: extractErrorInfo(error),
+        },
+      });
+      throw error;
+    }
+  }
+
+  <template>
+    <tr
+      data-component-id={{@component.id}}
+      class="d-admin-row__content admin-config-components__component-row"
+    >
+      <td class="d-admin-row__overview">
+        <LinkTo
+          class="d-admin-row__overview-name"
+          @route="adminCustomizeThemes.show"
+          @models={{array "themes" @component.id}}
+        >
+          {{@component.name}}
+        </LinkTo>
+        {{#if @component.remote_theme.authors}}
+          <div class="admin-config-components__author-name">{{i18n
+              "admin.config_areas.themes_and_components.components.by_author"
+              (hash name=@component.remote_theme.authors)
+            }}</div>
+        {{/if}}
+        {{#if @component.description}}
+          <div
+            class="d-admin-row__overview-about admin-config-components__description"
+          >
+            {{@component.description}}
+            {{#if @component.remote_theme.about_url}}
+              <a href={{@component.remote_theme.about_url}}>{{i18n
+                  "admin.config_areas.themes_and_components.components.learn_more"
+                }}</a>
+            {{/if}}
+          </div>
+        {{/if}}
+        {{#if this.hasUpdates}}
+          <div
+            class="d-admin-row__overview-about admin-config-components__update-available"
+          >
+            <b>{{i18n
+                "admin.config_areas.themes_and_components.components.update_available"
+              }}</b>
+          </div>
+        {{/if}}
+      </td>
+      <td class="d-admin-row__detail admin-config-components__parent-themes">
+        {{#if @component.parent_themes.length}}
+          {{this.parentThemesCell}}
+        {{else}}
+          <LinkTo
+            @route="adminCustomizeThemes.show"
+            @models={{array "themes" @component.id}}
+          >
+            {{i18n
+              "admin.config_areas.themes_and_components.components.add_to_theme"
+            }}
+          </LinkTo>
+        {{/if}}
+      </td>
+      <td class="d-admin-row__controls">
+        <div class="d-admin-row__controls-options">
+          <div class="d-admin-row__mobile-label">
+            {{i18n
+              "admin.config_areas.themes_and_components.components.enabled"
+            }}
+          </div>
+          <DToggleSwitch
+            @state={{this.enabled}}
+            class="admin-config-components__toggle"
+            disabled={{this.disableToggle}}
+            {{on "click" this.toggleEnabled}}
+          />
+          <DButton
+            class="admin-config-components__edit"
+            @label="admin.config_areas.themes_and_components.components.edit"
+            @route="adminCustomizeThemes.show"
+            @routeModels={{array "themes" @component.id}}
+          />
```

Comment:
I do wonder if this button is necessary when the name of the theme already links to go configuration page. When the theme component has not been added to a theme yet, we display 2 links and 1 button just to go to the theme component config page.

<img width="1081" alt="Screenshot 2025-04-04 at 4 18 55 PM" src="https://github.com/user-attachments/assets/fca54196-3311-4c78-9d08-da462208bc97" />

@ellaestigoy I wonder if there is a better UI pattern we can adopt here.

Reply:
Yeah, good point, it should follow the pattern we use in places like Plugin and Moderation. I’d keep the “Edit” button since it’s clearer that it links out to a different page, but I’ve removed the link from the theme name to make it less redundant.